File: notifica_crimes_frontend/lib/domain/models/ocorrencias/assalto.dart
import 'package:notifica_crimes_frontend/domain/models/ocorrencias/ocorrencia.dart';

class Assalto {

  final int qtdAgressores;
  final bool possuiArma;
  final bool tentativa;
  final String tipoArmaId;
  final List<String> tipoBensId;
  final Ocorrencia ocorrencia;

  Assalto({required this.qtdAgressores, required this.possuiArma, required this.tentativa, required this.tipoArmaId, required this.tipoBensId, required this.ocorrencia, });
  
 

}

File: NotificaCrimesBackEnd/AppNotificacoesCrimesCidade.Domain/Entities/Assalto.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace AppNotificacoesCrimesCidade.Domain.Entities
{
    public class Assalto
    {
        [Column("id")]
        public int Id { get; set; }

        [Column("qtd_agressores")]
        public int QtdAgressores { get; set; }

        [Column("possui_arma")]
        public bool PossuiArma { get; set; }

        [Column("id_ocorrencia")]
        public int OcorrenciaId { get; set; }

        public Ocorrencia Ocorrencia { get; set; }

        [Column("tentativa")]
        public bool Tentativa { get; set; }

        [Column("id_tipo_arma")]
        public int? TipoArmaId { get; set; }

        public TipoArma TipoArma { get; set; }

        public List<AssaltoTipoBem> AssaltoTipoBens { get; set; }
    }
}
